pkg/rid/application/isa.go

```diff
@@ -119,30 +118,32 @@ func (a *app) InsertISA(ctx context.Context, isa *ridmodels.IdentificationServic
 	// The following will automatically retry TXN retry errors.
 	err = a.Transactor.InTxnRetrier(ctx, func(repo repos.Repository) error {
 		var err error
-		cells := isa.Cells
-		if old != nil {
-			// TODO steeling, we should change this to a Custom type, to obfuscate
-			// some of these metrics and prevent us from doing the wrong thing.
-			cells = s2.CellUnionFromUnion(old.Cells, isa.Cells)
-			geo.Levelify(&cells)
-		}
+		// TODO steeling, we should change this to a Custom type, to obfuscate
+		// some of these metrics and prevent us from doing the wrong thing.
+		cells := s2.CellUnionFromUnion(old.Cells, isa.Cells)
+		geo.Levelify(&cells)
 		// UpdateNotificationIdxsInCells is done in a Txn along with insert since
 		// they are both modifying the db. Insert a susbcription alone does
 		// not do this, so that does not need to use a txn (in subscription.go).
 		subs, err = a.Transactor.UpdateNotificationIdxsInCells(ctx, cells)
 		if err != nil {
 			return err
 		}
-		ret, err = a.Transactor.InsertISA(ctx, isa)
-		if err != nil {
+		ret, err = a.Transactor.UpdateISA(ctx, isa)
+		switch {
+		case err != nil:
 			return err
+		case ret == nil:
+			// TODO: this should return a 404 or a 409, but the standard doesn't allow
+			// for a 404 on updates.
+			// When this changes, we can model after UnsafeDelete, and check the
+			// version param here.
+			return dsserr.AlreadyExists("this ISA was either updated or deleted")
```

Comment:
It doesn't make sense to return an error named "AlreadyExists" when updating something that exists -- even if it maps to the same error code (409), it should be called something like "InconsistentVersion".  I think the spirit would be 404 for missing ID and 409 for different version.  The standard doesn't specify 404 for this endpoint, but it also doesn't specify, e.g., 504 which is nevertheless a valid response under certain conditions.  The standard does not prohibit us from returning an appropriate error code even though it isn't listed.